scripts/generate_enums.py

```diff
@@ -0,0 +1,356 @@
+        enum_file.write("package roc\n\n")
+        enum_file.write(self.format_comment(enum_definition.doc, ""))
+        enum_file.write("//\n")
+        roc_prefix = self.name_prefixes[enum_definition.name]
+        go_prefix = to_camel_case(roc_prefix.lower().removeprefix('roc_').removesuffix('_'))
+        enum_file.write(f"//go:generate stringer -type {go_type_name} -trimprefix {go_prefix} -output {go_name}_string.go\n")
+        enum_file.write(f"type {go_type_name} int\n\n")
+
+        enum_file.write("const (\n")
+
+        for i, enum_value in enumerate(enum_values):
+            go_enum_value = to_camel_case(enum_value.name.lower().removeprefix('roc_'))
+
+            if i != 0:
+                enum_file.write("\n")
+            enum_file.write(self.format_comment(enum_value.doc, "\t"))
+            enum_file.write(f"\t{go_enum_value} {go_type_name} = {enum_value.value}\n")
+
+        enum_file.write(")\n")
+
+        enum_file.close()
+
+    def format_comment(self, doc: DocComment, indent: string):
+        indent_line = indent + "// "
+        doc_string = ""
+
+        for i, item in enumerate(doc.items):
+            if i != 0:
+                doc_string += indent_line.rstrip() + "\n"
+
+            text = self.doc_item_to_string(item)
+            for t in text.split("\n"):
+                lines = textwrap.wrap(t, width=80,
+                                      break_on_hyphens=False,
+                                      initial_indent=indent_line,
+                                      subsequent_indent=indent_line)
+                for line in lines:
+                    doc_string += line + "\n"
+
+        return doc_string
+
+    def doc_item_to_string(self, doc_item):
+        result = []
+        for item in doc_item:
+            t = item['type']
+            if t == "text":
+                result.append(item['text'])
+            elif t == "ref" or t == "code":
+                result.append(self.ref_to_string(item['text']))
+            elif t == "list":
+                ul = "\n"
+                for li in item['values']:
+                    ul += f" - {self.doc_item_to_string(li)}\n"
+                ul += "\n"
+                result.append(ul)
+            else:
+                print(f"unknown doc item type = {t}, consider adding it to doc_item_to_string")
+        return ' '.join(result).replace(" ,", ",").replace(" .", ".")
+
+    def ref_to_string(self, ref_value):
+        """
+        :param ref_value: enum_value or enum_type, e.g. roc_endpoint or ROC_INTERFACE_CONSOLIDATED
+        :return: go value
+        """
+        if ref_value.startswith("roc_"):
+            return to_camel_case(ref_value.removeprefix('roc_'))
+
+        return to_camel_case(ref_value.lower().removeprefix('roc_'))
+
+
+def parse_config_xml():
+    try:
+        tree = ElementTree.parse(CONFIG_FILE_PATH)
+        return tree.getroot()
+    except FileNotFoundError:
+        print(f"File not found: {CONFIG_FILE_PATH}")
```

Comment:
nit: convention is to print errors and warnings to stderr

Reply:
done